pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -1431,6 +1447,18 @@ func (gce *GCECloud) deleteTargetPool(name, region string, hc *compute.HttpHealt
 			return err
 		}
 	}
+	// Deletion of health checks is allowed only after the TargetPool reference is deleted
+	if hc != nil {
+		glog.Infof("Deleting health check %v", hc.Name)
+		if err := gce.DeleteHttpHealthCheck(hc.Name); err != nil {
+			glog.Warningf("Failed to delete health check %v: %v", hc, err)
+			return err
+		}
+	} else {
+		if err := gce.DeleteHttpHealthCheck(name); err == nil {
```

Comment:
can you clarify in a comment when/if we come through and cleanup this leak in the case of an error?